### ISSUE TITLE
Percentage encoded + in querystring gets lost

### ISSUE THREAD

**Issue by simonw** [simonw/datasette-publish-vercel#28]
Compare this example (on Cloud Run): https://latest.datasette.io/fixtures?sql=select+date%28%27now%27%2C+%22%2B30+days%22%29

<img width="482" alt="fixtures__select_date__now_____30_days__" src="https://user-images.githubusercontent.com/9599/90698913-ce91e280-e236-11ea-8d63-6b2d5d5a88f6.png">

To this example (on Vercel): https://til.simonwillison.net/til?sql=select+date%28%27now%27%2C+%22%2B30+days%22%29&p0=2020-04-23

<img width="472" alt="til__select_date__now_____30_days__" src="https://user-images.githubusercontent.com/9599/90698935-d9e50e00-e236-11ea-9b3c-67326cbb0aeb.png">

In both cases the `+` from the form was encoded as `%2B` in the URL.

**Comment by simonw** [comment]
This looks like a Vercel bug:

https://til.simonwillison.net/-/asgi-scope?sql=a%2Bb returns `'query_string': b'sql=a+b'`
https://til.simonwillison.net/-/asgi-scope?sql=a+b returns `'query_string': b'sql=a b'`

**Comment by simonw** [comment]
And https://til.simonwillison.net/-/asgi-scope?sql=a%252Bb with double-encoding `a%252Bb` produces `'query_string': b'sql=a%2Bb'`.

**Comment by simonw** [comment]
If it's a Vercel bug it will be in this code: https://github.com/vercel/vercel/blob/fcb14fa70e800ff2bf387120ebdaf65affea6661/packages/now-python/now_init.py#L261-L280
```python
            url = urlparse(unquote(payload['path']))
            query = url.query.encode()
            path = url.path

            scope = {
                'server': (headers.get('host', 'lambda'), headers.get('x-forwarded-port', 80)),
                'client': (headers.get(
                    'x-forwarded-for', headers.get(
                        'x-real-ip', payload.get(
                            'true-client-ip', ''))), 0),
                'scheme': headers.get('x-forwarded-proto', 'http'),
                'root_path': '',
                'query_string': query,
                'headers': [[k.lower().encode(), v.encode()] for k, v in headers.items()],
                'type': 'http',
                'http_version': '1.1',
                'method': payload['method'],
                'path': path,
                'raw_path': path.encode(),
            }
```

**Comment by simonw** [comment]
This commit is useful as it helps demonstrate how testing works for now-python: https://github.com/vercel/vercel/commit/8253e76ec0f0540f194e3077bffc2325cdfe629d

**Comment by simonw** [comment]
This fixture: https://github.com/vercel/vercel/blob/fcb14fa70e800ff2bf387120ebdaf65affea6661/packages/now-python/test/fixtures/12-asgi-function/index.py is exercised by this test: https://github.com/vercel/vercel/blob/8a68211cad8d2afae02ac1b85184e07d8911d47b/packages/now-python/test/fixtures/12-asgi-function/now.json#L10

**Comment by simonw** [comment]
Also possibly relevant: https://github.com/vercel/vercel/commit/8503af75baf94632500d20714c40bf2c623349f9#diff-9fd170c5d0171ce84976ebc317c15bcb is a commit titled "[now-python] Fix space encoding"

**Comment by simonw** [comment]
Just saw his one again in the wild - upgrading to latest Vercel builder in #34 may fix this.

**Comment by styfle** [comment]
Yep, fixed in #35

**Comment by simonw** [comment]
Yup, that fixes it: https://datasette-vercel-python-fix.vercel.app/fixtures?sql=select+date(%27now%27%2C+%22%2B30+days%22)
